CMS-Enterprise/mint-app
# [MINT-3166] data exchange cleanup

# MINT-3166

## Description
This PR looks to address feedback from other PRs that we didn't want to hold up merging to feature. It address some concerns in the code.
1. Status calculations
   a. These were returning incorrect statuses ( changing completed to in progress on accident), ability to set back in progress
 2. Bool data type not set as a pointer in database or GO
 3. Store methods following the old receiver paradgim, and not using sql util get/ select receivers.
 4. No unit tests
 5. Remove core task list section
 
 
 Note, this doesn't seek to address everything that could be cleaned up, only the sections noted above. Another PR might be needed later to finalize some changes.



## How to test this change

1.  Ensure tests pass
2. Update a data exchange approach with postman and confirm it works

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
